compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsAnydataUniqueVisitor.java

```diff
@@ -279,6 +279,11 @@ public Boolean visit(BRecordType type) {
                 return false;
             }
         }
+
+        if (!type.sealed && (type.restFieldType == null)) {
+            return false;
```

Comment:
There's a codecov comment? Shall we verify if this is covered via tests?

Reply:
+1.
Maybe the latest commit to updating type precedence for unresolved fields may have made this code redundant.